mooncake-common/etcd/etcd_wrapper.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+/*
+#include <stdint.h> 
+*/
+import "C"
+
+import (
+	"context"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"unsafe"
+)
+
+var globalClient *CClient
+
+type CClient struct {
+	client *clientv3.Client
+}
+
+//export NewEtcdClient
+func NewEtcdClient(endpoints *C.char) C.uintptr_t {
+	if globalClient != nil {
+		return C.uintptr_t(uintptr(unsafe.Pointer(globalClient)))
+	}
+
+	endpoint := C.GoString(endpoints)
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: 5 * time.Second,
+	})
+
+	if err != nil {
+		return 0
+	}
+
+	globalClient = &CClient{client: cli}
+	return C.uintptr_t(uintptr(unsafe.Pointer(globalClient)))
+}
+
+//export EtcdPutWrapper
+func EtcdPutWrapper(clientPtr C.uintptr_t, key *C.char, value *C.char) int {
+	if clientPtr == 0 {
+		return -1
+	}
+	client := *(**CClient)(unsafe.Pointer(&clientPtr))
+	k := C.GoString(key)
+	v := C.GoString(value)
+	_, err := client.client.Put(context.Background(), k, v)
+	if err != nil {
```

Comment:
return the error message to C side could be better, by a *C.char argument and `C.CString`.